plugins/orchestrator/src/components/WorkflowDefinitionsListComponent/WorkflowDefinitionsListComponent.tsx

```diff
@@ -27,26 +30,94 @@ type WorkflowsTableProps = {
 
 export const WorkflowsTable = ({ items }: WorkflowsTableProps) => {
   const orchestratorApi = useApi(orchestratorApiRef);
-
   const navigate = useNavigate();
   const definitionLink = useRouteRef(workflowDefinitionsRouteRef);
   const executeWorkflowLink = useRouteRef(executeWorkflowRouteRef);
   const editLink = useRouteRef(editWorkflowRouteRef);
+  const [data, setData] = useState<Row[]>([]);
 
   interface Row {
     id: string;
     name: string;
+    lastRun: string;
+    lastRunStatus: string;
+    type: WorkflowCategory;
+    components: string;
     format: string;
   }
 
-  const columns: TableColumn[] = [{ title: 'Name', field: 'name' }];
-  const data: Row[] = items.map(item => {
+  const columns: TableColumn[] = [
+    { title: 'Name', field: 'name' },
+    { title: 'Last run', field: 'lastRun' },
+    { title: 'Last run status', field: 'lastRunStatus' },
+    { title: 'Type', field: 'type' },
+    { title: 'Components', field: 'components' },
+  ];
+
+  const getInitFormState = () => {
+    const assessmentExist = !!items.find(
+      item =>
+        item.definition.annotations?.find(
+          annotation => annotation === ASSESSMENT_WORKFLOW_TYPE,
+        ),
+    );
     return {
-      id: item.definition.id,
-      name: item.definition.name ?? '',
-      format: extractWorkflowFormatFromUri(item.uri),
+      filtersOpen: true,
+      filters: assessmentExist
+        ? { Type: WorkflowCategory.ASSESSMENT }
+        : undefined,
     };
-  });
+  };
+  const [initTableState, __] = useState(getInitFormState());
+
+  const getInitialState = useCallback(() => {
```

Comment:
I think we can use `useMemo` here instead of `useCallback` (e.g. `const initialState = useMemo(...`).
As this callback does not have args, simply memorizing the initial state would be enough.
wdyt?

Reply:
yes! useMemo is better here